### PR TITLE
Dev/saars/no ilogger diagnostics docs

### PR DESCRIPTION
**Pull request by xiaomi7732**
Documentation refresh for the next version of AI.K8s.
Primary changes:
1. Replace ILogger with DiagnosticSource for self-diagnostics.
1. Update Console Application example to use Application Insights ILogger provider.

Do NOT merge until the new package is released.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,26 +1,56 @@
...
+
+3. Build and run the app in containers managed by Kubernetes.
 
-* For **ASP.NET Core** Application: Refer [Getting Started](https://github.com/Microsoft/ApplicationInsights-Kubernetes/wiki/Getting-Started-for-ASP.NET-Core-Applications) for a simple walkthrough.
+### Walk-through
 
-* For **.NET Core** Application: Refer [Getting Started](https://github.com/Microsoft/ApplicationInsights-Kubernetes/wiki/Getting-Started-for-.NET-Core-Applications) for a simple walkthrough.
+Both **ASP.NET Core** and **.NET Core** applications are supported.
+
+* For **ASP.NET Core** Application: Refer [Getting Started](https://github.com/Microsoft/ApplicationInsights-Kubernetes/wiki/Getting-Started-for-ASP.NET-Core-Applications) for a simple walk-through.
```
Comment (karolz-ms) [inline README.md]: "refer to..." 

(same applies to the .NET Core case)

File: README.md [hunk 2]
```diff
@@ -29,13 +59,15 @@ We support **ASP.NET Core** application as well as **.NET Core** application.
 Customize configurations are supported for `v1.0.2+`. There are several ways to customize the settings. For example:
```
Comment (karolz-ms) [inline README.md]: It is not clear what the version number refers to. Suggest something like

Custom configuration are supported starting with version 1.0.2 of the `ApplicationInsights-Kubernetes` package.

File: README.md [hunk 3]
```diff
@@ -29,13 +59,15 @@ We support **ASP.NET Core** application as well as **.NET Core** application.
 Customize configurations are supported for `v1.0.2+`. There are several ways to customize the settings. For example:
 
 1. By the code:
```
Comment (karolz-ms) [inline README.md]: Suggest "Using code...", "using environment variable..." etc

File: README.md [hunk 4]
```diff
@@ -47,40 +79,48 @@ Customize configurations are supported for `v1.0.2+`. There are several ways to
...
 
-    | Key                   | Value/Types | Default Value | Description                                                                                            |
-    |-----------------------|-------------|---------------|--------------------------------------------------------------------------------------------------------|
-    | InitializationTimeout | TimeSpan    | 00:02:00      |  Maximum time to wait for spinning up the container. Accepted format: [d.]hh:mm:ss[.fffffff]. |
+    | Key                   | Value/Types | Default Value | Description                                                                                  |
+    | --------------------- | ----------- | ------------- | -------------------------------------------------------------------------------------------- |
+    | InitializationTimeout | TimeSpan    | 00:02:00      | Maximum time to wait for spinning up the container. Accepted format: [d.]hh:mm:ss[.fffffff]. |
 
 The configuration uses with the conventions in ASP.NET Core. Refer [Configuration in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-2.1) for more information.
```
Comment (karolz-ms) [inline README.md]: The configuration uses ASP.NET Core conventions. Refer to...

File: README.md [hunk 5]
```diff
@@ -47,40 +79,48 @@ Customize configurations are supported for `v1.0.2+`. There are several ways to
...
 
 ### Next step
+
 Profile your application for performance improvement using [Application Insights Profiler for Linux](https://github.com/Microsoft/ApplicationInsights-Profiler-AspNetCore).
 
 ## Contributing
+
 ### Report issues
+
 Please file bug, discussion or any other interesting topics in [issues](https://github.com/Microsoft/ApplicationInsights-Kubernetes/issues) on github.
```
Comment (karolz-ms) [inline README.md]: GitHub

File: README.md [hunk 6]
```diff
@@ -47,40 +79,48 @@ Customize configurations are supported for `v1.0.2+`. There are several ways to
...
 
 ### Trouble Shooting
```
Comment (karolz-ms) [inline README.md]: Troubleshooting

File: README.md (hunk 1 above)
Comment (karolz-ms) [reply README.md]: ... and in other places in this document

File: docs/SelfDiagnostics.MD [hunk 7]
```diff
@@ -0,0 +1,65 @@
+# [Advanced] How to enable self diagnostics for ApplicationInsights.Kubernetes
+
+## Enable self-diagnostics
+
+We understand ApplicationInsights.Kubernetes is far from perfect. Enabling self-diagnostics will help find [issues](https://github.com/microsoft/ApplicationInsights-Kubernetes/issues) easier. Please feel free to open issues and we will be glad to look into them.
```
Comment (karolz-ms) [inline docs/SelfDiagnostics.MD]: The first sentence suggests our confidence in this package is very low.

Consider saying something along the lines of:
1. Enabling self-diagnostics will help troubleshoot issues and determine whether the problem is with service code/configuration, or with the `ApplicationInsights.Kubernetes` package itself.
1. If problem with AI.K package is suspected, please open an issue on GitHub and we will look into it.

File: docs/SelfDiagnostics.MD [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+
+Application Insights for Kubernetes is instrumented with DiagnosticsSource. To see the logs, create a diagnostic source listener that listens to `Microsoft.ApplicationInsights.Kubernetes.Debugging.Inspect.Observer`. The observer(DiagnosticSource) will emits logs in different levels.
```
Comment (karolz-ms) [inline docs/SelfDiagnostics.MD]: OK, this is a good example of bad naming that makes things confusing.

Assuming `Inspect` becomes `XxxDiagnosticSource`, the property should be called `Listener`. You should then instruct the user to create an `Observer` that subscribes to the `Listener` to consume logs from the AI.K package

File: docs/SelfDiagnostics.MD [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+
+To make diagnostic easier, a default listener that outputs logs into the console is provided and and be enabled like it below before `AddApplicationInsightsKubernetesEnricher` is called:
+
+```csharp
+var listener = InspectListener.Instance;
```
Comment (karolz-ms) [inline docs/SelfDiagnostics.MD]: "Invert" the names please

```csharp
var observer = new AIKObserver();  // Don't see a compelling reason why this should be a singleton
observer.SetMinimumLevel(DiagnosticLevel.Trace);
AIKDiagnosticSource.Instance.Listener.SubscribeWithAdapter(observer);
```

File: docs/SelfDiagnostics.MD [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+listener.SetMinimumLevel(InspectLevel.Trace);
+Inspect.Instance.Observer.SubscribeWithAdapter(listener);
+```
+
+When the listener above is enabled, you will see diagnostic logs like this:
+
+```shell
+...
+[Debug] Application Insights Kubernetes injected the service successfully.
```
Comment (karolz-ms) [inline docs/SelfDiagnostics.MD]: Consider

- adding timestamp to logs
- somehow indicating the log comes from the AIKObserver (as opposed to all the other noise written to std output stream).

File: README.md [hunk 11]
```diff
@@ -1,26 +1,56 @@
...
+    {
+        ...
+        services.AddApplicationInsightsTelemetry();
+        services.AddApplicationInsightsKubernetesEnricher();
+        services.AddMvc();
+        ...
+    }
+    ```
+
+3. Build and run the app in containers managed by Kubernetes.
```
Comment (karolz-ms) [inline README.md]: Is it the case that the example will not work if Application Insights instrumentation key is not set? If so, I would mention it, as it is a common mistake.

File: docs/SelfDiagnostics.MD (hunk 10 above)
Comment (xiaomi7732) [reply docs/SelfDiagnostics.MD]: Will consider the implementation separately.

File: README.md (hunk 4 above)
Comment (xiaomi7732) [reply README.md]: Yes

File: README.md (hunk 5 above)
Comment (xiaomi7732) [reply README.md]: Yup

File: README.md (hunk 6 above)
Comment (xiaomi7732) [reply README.md]: Sure.

File: docs/SelfDiagnostics.MD (hunk 7 above)
Comment (xiaomi7732) [reply docs/SelfDiagnostics.MD]: Good suggestions!

File: docs/SelfDiagnostics.MD (hunk 9 above)
Comment (xiaomi7732) [reply docs/SelfDiagnostics.MD]: Yup yup.

File: README.md (hunk 11 above)
Comment (xiaomi7732) [reply README.md]: Updated.

**Pull request merged**